theano/gpuarray/dnn.py

```diff
@@ -2732,6 +2732,191 @@ def local_abstractconv3d_cudnn_graph(op, context_name, inputs, outputs):
     return [rval]
 
 
+class GpuDnnSpatialTfDesc(COp):
+
+    """
+    This Op builds a spatial transformer descriptor for use in spatial transformer network
+    operations.
+    """
+
+    __props__ = ('dimensions', 'precision')
+    params_type = ParamsType(nimages=int_t, nchannels=int_t, height=int_t, width=int_t,
+                             nb_dims=int_t, precision=cudnn.cudnnDataType_t)
+
+    def c_headers(self):
+        return ['cudnn.h', 'cudnn_helper.h']
+
+    def c_header_dirs(self):
+        return [os.path.dirname(__file__), config.dnn.include_path]
+
+    def c_libraries(self):
+        return ['cudnn']
+
+    def c_lib_dirs(self):
+        return [config.dnn.library_path]
+
+    def do_constant_folding(self, node):
+        return False
+
+    def __init__(self, dimensions, precision="float32"):
+        COp.__init__(self, ["c_code/spatialtf_desc.c"], "APPLY_SPECIFIC(spatialtf_desc)")
+
+        # dimensions must have at least width and height
+        assert len(dimensions) >= 2
```

Comment:
Absolutely, even a simple `reshape((1, 1, width, height))` is enough to provide the proper data layout. I've modified the assert to check if the img has four dimensions.